src/main/java/uk/gov/hmcts/ccd/CachingConfiguration.java

```diff
@@ -37,11 +81,43 @@ private void configCaches(int definitionCacheTTL, Config config) {
         config.addMapConfig(newMapConfig("userRolesCache", definitionCacheTTL));
     }
 
+    private void warmUp(final HazelcastInstance hazelcastInstance) {
+        IMap<String, CaseType> caseTypeDefinitionsCache = hazelcastInstance.getMap("caseTypeDefinitionsCache");
```

Comment:
I don't think this is going to work. We are trying to fetch from def store all the case type definitions at once. Consider that at the moment retrieving 1 definition
is a very expensive operation. Which takes about 10 seconds. I'm worried this single call will kill Def Store. 
Also with this approach we introduced a risky endpoint in Def Store which fetches all the definitions. We already had that endpoint in the past,
and I removed it to make sure no one attempts to do something like that. Which can kill Def Store db.
I think a better approach would be to retrieve from the Def Store only the ids of the case types. Then in Data Store we can do a looping logic which 
retrieves the case types one by one. 
If you do this, you can even avoid adding all this logic here in this class, you can just call the existing getCaseType, which already does caching automatically.
We could also customise the fetching to say for example wait 1 second between each single case type fetch. To avoid overloading the Def Store.

Reply:
Sure will change.

Reply:
cheers 👍

Reply:
I cannot call the existing getCaseType(id) as I do not have request scope here. Remember this is an app startup so I have to build the headers for the cache warm up user manually.